### PR TITLE
feat(MCM): add new MCM endpoints

### PR DESCRIPTION
**Pull request by Nagriar**

### CODE REVIEW

File: src/AlgoliaSearch/Client.php [hunk 1]
```diff
@@ -404,6 +404,217 @@ public function getLogs($offset = 0, $length = 10, $type = 'all', $requestHeader
...
+        );
+    }
+
+    /**
+     * Remove a userID from the mapping
+     * @return an object containing a "deletedAt" attribute
+     *
+     * @throws AlgoliaException
+     */
+    public function RemoveUserID($userID, $requestHeaders = array())
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Client.php]: "R" should be lowercased

File: src/AlgoliaSearch/Client.php [hunk 2]
```diff
@@ -404,6 +404,217 @@ public function getLogs($offset = 0, $length = 10, $type = 'all', $requestHeader
...
+     *     ),
+     *     "nbHits" => 0
+     *     "page" => 0,
+     *     "hitsPerPage" => 20
+     * ).
+     *
+     * @return mixed
+     * @throws AlgoliaException
+     */
+    public function searchUserIDs($query, $clusterName = null, $page = null, $hitsPerPage = null, $requestHeaders = array())
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Client.php]: We should probably merge the 4 first params together.

Comment (Nagriar) [reply src/AlgoliaSearch/Client.php]: I used the same format than `searchSynonyms` https://github.com/algolia/algoliasearch-client-php/blob/master/src/AlgoliaSearch/Index.php#L1397

File: src/AlgoliaSearch/Client.php [hunk 3]
```diff
@@ -404,6 +404,215 @@ public function getLogs($offset = 0, $length = 10, $type = 'all', $requestHeader
...
+            $params['hitsPerPage'] = $hitsPerPage;
+        }
+
+        return $this->request(
+            $this->context,
+            'POST',
+            '/1/clusters/mapping/search',
+            null,
+            $params,
+            $this->context->readHostsArray,
```
Comment (rayrutjes) [inline src/AlgoliaSearch/Client.php]: Should this be writeHosts maybe?

Comment (rayrutjes) [reply src/AlgoliaSearch/Client.php]: It's good as is

Comment (Nagriar) [reply src/AlgoliaSearch/Client.php]: It's expected, this operation can be performed on a read-only node

**Pull request merged**